storage/client/client.go

```diff
@@ -53,7 +53,6 @@ type StorageClient struct {
 	networkConfig  *oasisConfig.Network
 
 	blockCache *ristretto.Cache
```

Comment:
txCache is removed because the current trivial implementation is wrong due to the possibility of duplicate transaction hashes.

Also, this has been implemented very long ago, and maybe a bit prematurely, since, e.g., for runtime txes, we don't have any such caches.

I propose to drop it at this point, and we can revisit in the future if this query ends up problematic.